bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeUPNPDiscoveryParticipant.java

```diff
@@ -0,0 +1,162 @@
+            }
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return HueBridgeHandler.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
+        if (isAutoDiscoveryEnabled) {
+            DeviceDetails details = device.getDetails();
+            ThingUID uid = getThingUID(device);
+            if (details != null && uid != null) {
+                URL baseUrl = details.getBaseURL();
+                String serialNumber = details.getSerialNumber();
+                String label = details.getFriendlyName();
+                String modelName = EXPECTED_MODEL_NAME_PREFIX;
+                ModelDetails modelDetails = details.getModelDetails();
+                if (modelDetails != null && modelDetails.getModelName() != null
+                        && modelDetails.getModelNumber() != null) {
+                    modelName = String.format("%s (%s)", modelDetails.getModelName(), modelDetails.getModelNumber());
+                }
+                if (baseUrl != null && serialNumber != null && !serialNumber.isBlank() && label != null) {
+                    return DiscoveryResultBuilder.create(uid) //
+                            .withProperties(Map.of( //
+                                    HOST, baseUrl.getHost(), //
+                                    PORT, baseUrl.getPort(), //
+                                    PROTOCOL, baseUrl.getProtocol(), //
+                                    Thing.PROPERTY_MODEL_ID, modelName, //
+                                    Thing.PROPERTY_SERIAL_NUMBER, serialNumber.toLowerCase())) //
+                            .withLabel(label) //
+                            .withRepresentationProperty(Thing.PROPERTY_SERIAL_NUMBER) //
+                            .build();
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(RemoteDevice device) {
+        DeviceDetails details = device.getDetails();
+        if (details != null) {
+            String serialNumber = details.getSerialNumber();
+            ModelDetails modelDetails = details.getModelDetails();
+            if (serialNumber != null && !serialNumber.isBlank() && modelDetails != null) {
+                String modelName = modelDetails.getModelName();
+                // Model name has the format "Philips hue bridge <year>" with <year> being 2012
+                // for a hue bridge V1 or 2015 for a hue bridge V2.
+                if (modelName != null && modelName.startsWith(EXPECTED_MODEL_NAME_PREFIX)) {
+                    boolean ignored = false;
+                    try {
+                        int year = Integer.parseInt(modelName.substring(EXPECTED_MODEL_NAME_PREFIX.length()));
+                        // The bridge is ignored if year is lower than 2015
+                        ignored = year < 2015;
```

Comment:
Shouldn't it be the opposite, i.e. newer bridges should be ignored?

I'm wondering if any of this logic is necessary since bridge V2 with new firmware can't be discovered by UPnP since Q2 2022, that seems like effective way of ignoring? Reference: https://developers.meethue.com/new-hue-api/

Reply:
Oups, you're right and my test should not have worked so I will have to test it again seriously.

@andrewfg shows this morning using Device Spy that UPnP is still available for bridge V2.

Reply:
> @andrewfg shows this morning using Device Spy that UPnP is still available for bridge V2.

Oh, I missed that, I need to catch up. Will test with my bridge also.

Reply:
I tested with this line added after line 140:

```java
                    logger.info("Discovered Hue bridge: {}, {}, {}", modelDetails.getModelName(),
                            modelDetails.getModelNumber(), modelDetails.getModelDescription());
```

Result:
```
2023-05-01 21:30:38.087 [INFO ] [ry.HueBridgeUPNPDiscoveryParticipant] - Discovered Hue bridge: Philips hue bridge 2015, BSB002, Philips hue Personal Wireless Lighting
2023-05-01 21:30:38.088 [INFO ] [ry.HueBridgeUPNPDiscoveryParticipant] - Discovered Hue bridge: Philips hue bridge 2015, BSB002, Philips hue compatible Personal Wireless Lighting
```

So I can confirm that UPnP is still enabled with firmware version 1957113050. And I can also confirm that no discovery result is created by this code.